### PR TITLE
add hw/sw information to executor metadata

### PR DESCRIPTION
**Pull request by maxkozlovsky**

### CODE REVIEW

File: crates/indexify_internal_api/src/lib.rs [hunk 1]
```diff
@@ -1146,12 +1146,58 @@ impl Default for ContentMetadata {
...
+#[derive(Debug, Clone, Serialize, Deserialize, Default)]
 pub struct ExecutorMetadata {
     pub id: String,
-    pub last_seen: u64,
     pub addr: String,
     pub extractors: Vec<ExtractorDescription>,
+    #[serde(default)]
+    pub os_type: String,
+    #[serde(default)]
+    pub os_version: VersionInfo,
```
Comment (diptanu) [inline crates/indexify_internal_api/src/lib.rs:1163]: This probably needs to be a string or some kind of type which denotes a range.

File: crates/indexify_internal_api/src/lib.rs [hunk 2]
```diff
@@ -1146,12 +1146,58 @@ impl Default for ContentMetadata {
...
+    #[serde(default)]
+    pub python_version: VersionInfo,
+    #[serde(default)]
+    pub memory: u64,
+    #[serde(default)]
+    pub num_cpus: u32,
+    #[serde(default)]
+    pub gpu_memory: u64,
```
Comment (diptanu) [inline crates/indexify_internal_api/src/lib.rs:1171]: Lets rename this to gpu_memory_mb

File: crates/indexify_internal_api/src/lib.rs [hunk 3]
```diff
@@ -1146,12 +1146,58 @@ impl Default for ContentMetadata {
...
     pub addr: String,
     pub extractors: Vec<ExtractorDescription>,
+    #[serde(default)]
+    pub os_type: String,
+    #[serde(default)]
+    pub os_version: VersionInfo,
+    #[serde(default)]
+    pub python_version: VersionInfo,
+    #[serde(default)]
+    pub memory: u64,
```
Comment (diptanu) [inline crates/indexify_internal_api/src/lib.rs:1167]: Lets rename this to memory_mb

File: protos/coordinator_service.proto [hunk 4]
```diff
@@ -229,10 +229,21 @@ message Content {
...
 message RegisterExecutorRequest {
     string executor_id = 1;
     string addr = 2;
     repeated Extractor extractors = 3;
+    string os_type = 4;
+    VersionInfo os_version = 5;
+    VersionInfo python_version = 6;
+    uint32 num_cpus = 7;
+    uint64 memory = 8;
+    uint64 gpu_memory = 9;
```
Comment (diptanu) [inline protos/coordinator_service.proto:246]: Rename the memory vars to have _mb

File: src/state/store/compat/mod.rs [hunk 5]
```diff
@@ -10,12 +10,15 @@ use crate::state::{NodeId, StateMachineColumns};
 mod convert_v1;
 mod convert_v2;
 mod convert_v3;
+mod convert_v4;
```
Comment (diptanu) [inline src/state/store/compat/mod.rs:13]: Note for later : Lets remove 1, 2 and 3. We need to maintain compatibility with last released version.

**Review by diptanu — CHANGES_REQUESTED** [review]
Left some comments.

File: crates/indexify_internal_api/src/lib.rs (hunk 1 above)
Comment (maxkozlovsky) [reply crates/indexify_internal_api/src/lib.rs:1163]: the version is concrete for executor instance, why would it be a range?

File: crates/indexify_internal_api/src/lib.rs (hunk 2 above)
Comment (maxkozlovsky) [reply crates/indexify_internal_api/src/lib.rs:1171]: the values are in bytes it is simpler to not use specific units

Comment (maxkozlovsky) [reply crates/indexify_internal_api/src/lib.rs]: added type

**Pull request merged**